bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -548,14 +548,24 @@ private boolean removeIfLedgerNotExists(EntryLogMetadata meta) throws EntryLogMe
     @VisibleForTesting
     void doCompactEntryLogs(double threshold, long maxTimeMillis) throws EntryLogMetadataMapException {
         LOG.info("Do compaction to compact those files lower than {}", threshold);
+        double tmpGcEntryLogSizeRatio = conf.getGcEntryLogSizeRatio() <= 0 ? 0 : conf.getGcEntryLogSizeRatio();
+        final double gcEntryLogSizeRatio = tmpGcEntryLogSizeRatio > 1 ? 1.0 : tmpGcEntryLogSizeRatio;
+        if (gcEntryLogSizeRatio > 0.5) {
+            LOG.warn("Configured gcEntryLogSizeRatio: {}, updated gcEntryLogSizeRatio: {} gratter than 0.5, "
+                + "which means any entryLog file size less than {} will be compacted and it will bring heavy pressure "
+                + "on garbage collection.", conf.getGcEntryLogSizeRatio(), gcEntryLogSizeRatio,
+                gcEntryLogSizeRatio * conf.getEntryLogSizeLimit());
+        }
 
         final int numBuckets = 10;
         int[] entryLogUsageBuckets = new int[numBuckets];
         int[] compactedBuckets = new int[numBuckets];
 
         ArrayList<LinkedList<Long>> compactableBuckets = new ArrayList<>(numBuckets);
+        ArrayList<LinkedList<Long>> smallFilesCompactableBuckets = new ArrayList<>(numBuckets);
```

Comment:
I think you can avoid addition of another type of buckets.
Have something like useTargetEntryLogSizeForGc config;

Replace
```java
calculateUsageIndex(numBuckets, meta.getUsage());
```

with something like
```java
double usage = meta.getUsage();

if (conf.getUseTargetEntryLogSizeForGc()) {
  usage = (double) meta.getRemainingSize() / Math.max(meta.getTotalSize(), conf.getEntryLogSizeLimit());
}
```
and the rest will happen naturally.

As an improvement, transactional compaction can fallback to a regular one if the estimated size of entry log after compaction is below some limit.

Reply:
@dlg99 Good suggestion. I updated the code with `useTargetEntryLogSizeForGc` flag.  Please help take a look, thanks.

> As an improvement, transactional compaction can fallback to a regular one if the estimated size of entry log after compaction is below some limit.

For this one, I will open a new PR to handle it.